src/__tests__/extensions/surveys/survey-popup.test.tsx

```diff
@@ -1,34 +1,62 @@
 import '@testing-library/jest-dom'
-import { cleanup, fireEvent, render, screen } from '@testing-library/preact'
+import { cleanup, fireEvent, render, screen, waitFor } from '@testing-library/preact'
 import { SurveyPopup } from '../../../extensions/surveys'
+import * as surveyUtils from '../../../extensions/surveys/surveys-extension-utils' // Import all utils
 import { Survey, SurveyQuestionType, SurveyType } from '../../../posthog-surveys-types'
+import * as uuid from '../../../uuidv7' // Import uuidv7
+
+// Mock the utility functions
+jest.mock('../../../extensions/surveys/surveys-extension-utils', () => ({
+    ...jest.requireActual('../../../extensions/surveys/surveys-extension-utils'), // Keep original implementations for non-mocked parts
+    getInProgressSurveyState: jest.fn(),
+    // No need to mock set/clear directly if we verify send/dismiss event calls
+    // setInProgressSurveyState: jest.fn(),
+    // clearInProgressSurveyState: jest.fn(),
+    sendSurveyEvent: jest.fn(),
+    dismissedSurveyEvent: jest.fn(),
+}))
+
+// Mock uuidv7
+jest.mock('../../../uuidv7')
+
+// Mock PostHog instance needed by event handlers
+const mockPosthog = {
+    capture: jest.fn(),
+    get_session_replay_url: jest.fn().mockReturnValue('http://example.com/replay'),
+}
 
 describe('SurveyPopup', () => {
-    // Create a basic mock survey for testing
     const mockSurvey: Survey = {
-        id: 'test-survey',
-        name: 'Test Survey',
-        description: 'A test survey',
+        id: 'test-survey-partial',
+        name: 'Test Partial Survey',
+        description: 'A test survey for partial responses',
         type: SurveyType.Popover,
+        enable_partial_responses: true, // Enable partial responses for testing
```

Comment:
should this be a different test then since the behaviour changes based on the `enable_partial_responses`?

Reply:
not a different test, but yes should be a diffenret mock. [updated now](https://github.com/PostHog/posthog-js/pull/1888/commits/098444e1cd3f0310040c57ac05c8bd188b3bd0b1)